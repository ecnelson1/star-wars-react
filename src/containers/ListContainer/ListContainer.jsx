import React, {useState, useEffect} from 'react';
import CharacterList from '../../components/character/CharacterList';
import { getCharactersBySearch} from '../../services/api-utils';


const ListContainer = () => {
    const [search, setSearch] = useState('')
    const [characters, setCharacters] = useState([]);
    const [currentPage, setCurrentPage] = useState(1);
    const [pages, setPages] = useState();
    const [loading, setLoading] = useState(true)
   

    const handleNext = () => {
        if(currentPage < pages){
      setCurrentPage(currentPage + 1 );
        } else {
            return;
        }
    };
    const handlePrev = () => {
        if(currentPage > 1){
      setCurrentPage(currentPage - 1);
        } else {
            return;
        }
    };

    const handleSearchChange = (event) =>{
        setSearch(event.target.value)
    }

    const handleSubmit= (event) =>{
        event.preventDefault();
        getCharactersBySearch(search, currentPage)
        .then((res)=>setPages(res.count/10))
        .then(setLoading(false));
    }

      useEffect(() => {
       getCharactersBySearch(search, currentPage)
         .then((res) => {
           setCharacters(res.sorted);
           const count = res.count;
           setPages(count / 10);
         })
         .finally(setLoading(false));
     }, [search, currentPage]);

    return (
      <div title='list-container'>
        <form className="search-form" onSubmit={handleSubmit}>
          <input
            placeholder="Enter character name to search"
            value={search}
            onChange={handleSearchChange}
          />
        </form>
        {loading ? (
          <h1>LOADING...</h1>
        ) : (
          <>
            <CharacterList characters={characters} />{" "}
            <button onClick={handlePrev}>Prev Page</button>
            <button onClick={handleNext}>Next Page</button>
          </>
        )}
      </div>
    );
};

export default ListContainer;
