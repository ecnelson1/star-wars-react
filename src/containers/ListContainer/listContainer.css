.avg{
    display: grid;
    color: lawngreen;
    height: 20px;
    width: 100%;
    margin: 5px;
    padding: 15px;
}
.avgH{
    display: grid;
    color: lawngreen;
    height: 20px;
    margin: 5px;
    
}
.avgM{
    display: grid;
    color: lawngreen;
    height: 20px;
    margin: 5px;
    
}
.pageNav{
margin: 20px;
padding: 20px;
}
