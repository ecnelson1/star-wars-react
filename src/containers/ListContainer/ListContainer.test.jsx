import React from 'react';
import { render, screen } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node'
import ListContainer from './ListContainer'
import mockData from '../../../fixtures/mock-data.json'

const server = setupServer(rest.get('https://swapi.dev/api/people', (req, res, ctx)=> {
    return res(ctx.json(mockData))
}));

describe('ListContainer', () => {
    beforeAll(()=> server.listen());
    afterAll(()=> server.close());

    it('renders a loading element', async () => {
        render(<ListContainer/>);
        const loading = screen.getByText('LOADING...');
        expect(loading).toMatchSnapshot();
       
    });
    
    it('renders a list of Star Wars characters', async ()=>{
        render(<ListContainer/>);
        const ul = await screen.findByRole('list', {name: 'character-list'});
        expect(ul).toMatchSnapshot();
    });
});
