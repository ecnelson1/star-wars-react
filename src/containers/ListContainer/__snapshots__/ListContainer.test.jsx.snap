// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContainer renders a list of Star Wars characters 1`] = `
<ul
  aria-label="character-list"
/>
`;
